packages/sku/src/services/vite/createConfig.ts

```diff
@@ -0,0 +1,68 @@
+import { createRequire, builtinModules } from 'node:module';
+import type { InlineConfig } from 'vite';
+
+import react from '@vitejs/plugin-react';
+import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
+
+import type { SkuContext } from '@/context/createSkuContext.js';
+
+const require = createRequire(import.meta.url);
+
+const clientEntry = require.resolve('./entries/vite-client.jsx');
+
+export const createViteConfig = ({
+  skuContext,
+  configType = 'client',
+  plugins = [],
+}: {
+  skuContext: SkuContext;
+  configType?: 'client';
+  plugins?: InlineConfig['plugins'];
+}) => {
+  const outDir = {
+    client: 'dist',
+  };
+
+  return {
+    root: process.cwd(),
+    plugins: [
+      react(),
+      vanillaExtractPlugin({
+        unstable_pluginFilter: (plugin) =>
+          !plugin.name.startsWith('vite:react-'),
+      }),
+      ...plugins,
+    ],
+    resolve: {
+      alias: {
+        __sku_alias__clientEntry: skuContext.paths.clientEntry,
+        __sku_alias__serverEntry: skuContext.paths.serverEntry,
+        __sku_alias__renderEntry: skuContext.paths.renderEntry,
+      },
+    },
+    define: {
+      'process.env': process.env,
```

Comment:
Vite already has [its own envar conventions](https://vite.dev/guide/env-and-mode.html#env-variables-and-modes), so IMO the only envar we should expose here is `process.env.NODE_ENV` since braid relies on it (and it's documented sku feature too).